Tests/Get-BMDeployment.Tests.ps1

```
@@ -212,9 +240,177 @@ Describe 'Get-BMDeployment' {
+        ThenShouldNotThrowErrors
+        ThenDeploymentShouldNotBeReturned $deployment
+        ThenDeploymentShouldBeReturned $deployment2
+        ThenDeploymentShouldNetBeReturned $deployment3
+        ThenTotalDeploymentsReturned 1
+    }
+
+    It 'should get a deployment by status' {
+        $build = GivenReleaseBuild $script:releaseAll
+        $build2 = GivenReleaseBuild $script:releaseRelease
+        $deployment = GivenDeployment $build
+        $deployment2 = GivenDeployment $build -Stage 'Testing'
+        $deployment3 = GivenDeployment $build2
+        WhenGettingBMDeployment -Status 'failed'
+        ThenShouldNotThrowErrors
+        ThenDeploymentShouldNotBeReturned $deployment
+        ThenDeploymentShouldBeReturned $deployment2
+        ThenDeploymentShouldNetBeReturned $deployment3
+        ThenTotalDeploymentsReturned 1
+    }
+
+    It 'should get by multiple parameters' {
+        $build = GivenReleaseBuild $script:releaseAll
+        $build2 = GivenReleaseBuild $script:releaseRelease
+        $deployment = GivenDeployment $build
+        $deployment2 = GivenDeployment $build -Stage 'Testing'
+        $deployment3 = GivenDeployment $build2
+        WhenGettingBMDeployment -Release $script:releaseAll -PipelineStageName 'Testing'
+        ThenShouldNotThrowErrors
+        ThenDeploymentShouldNotBeReturned $deployment
+        ThenDeploymentShouldBeReturned $deployment2
+        ThenDeploymentShouldNetBeReturned $deployment3
+        ThenTotalDeploymentsReturned 1
+    }
+
+    It 'should find no deployments' {
+        $build = GivenReleaseBuild $script:releaseAll
+        $build2 = GivenReleaseBuild $script:releaseRelease
+        $deployment = GivenDeployment $build
+        $deployment2 = GivenDeployment $build -Stage 'Testing'
+        $deployment3 = GivenDeployment $build2
+        WhenGettingBMDeployment -Release $script:releaseAll -PipelineStageName 'Testing' -Application $script:app
+        ThenShouldNotThrowErrors

```

This test is wrong. It should throw an error and not return any deployments.